internal/pkg/cli/task_run.go

```diff
@@ -91,9 +109,50 @@ func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
 
 		deployer: cloudformation.New(sess),
 	}
+
+	opts.configureRuntimeOpts = func() error {
+		if err := opts.configureRepository(provider); err != nil {
+			return err
+		}
+		// TODO: configure runner
+		return nil
+	}
+
 	return &opts, nil
 }
 
+func (o *runTaskOpts) configureRepository(provider sessionProvider) error {
+	var registry repository.Registry
+	if o.env != "" {
+		env, err := o.targetEnv()
+		if err != nil {
+			return err
+		}
+
+		sess, err := provider.FromRole(env.ManagerRoleARN, env.Region)
+		if err != nil {
+			return fmt.Errorf("get session from role %s and region %s: %w", env.ManagerRoleARN, env.Region, err)
+		}
+
+		registry = ecr.New(sess)
+	} else {
+		sess, err := provider.Default()
+		if err != nil {
+			return fmt.Errorf("get default session: %w", err)
+		}
+		registry = ecr.New(sess)
+	}
```

Comment:
maybe sth like this?
```suggestion
	var sess *session.Session
	if o.env != "" {
		env, err := o.targetEnv()
		if err != nil {
			return err
		}

		sess, err := provider.FromRole(env.ManagerRoleARN, env.Region)
		if err != nil {
			return fmt.Errorf("get session from role %s and region %s: %w", env.ManagerRoleARN, env.Region, err)
		}
	} else {
		sess, err := provider.Default()
		if err != nil {
			return fmt.Errorf("get default session: %w", err)
		}
	}
	registry := ecr.New(sess)
```

Reply:
That would requires an import from aws-sdk-go/session, which I was trying to avoid. Do you think it's okay to have that dependency here?

Reply:
it looks ok to depend on aws-sdk-go to me because there's no chance it would cause cyclic reference issue. But it's up to you.

Reply:
I felt like we were trying to avoid explicit dependencies on aws-sdk-go from cli package... if that's not true I totally agree that I should just instantiate registry outside of the if-else block.

Reply:
:shipit: I think session is totally ok since we need to get the credentials in the `cli` pkg!